lib/suppressions.cpp

```diff
@@ -395,16 +400,15 @@ std::string Suppressions::Suppression::getText() const
 bool Suppressions::isSuppressed(const Suppressions::ErrorMessage &errmsg, bool global)
 {
     const bool unmatchedSuppression(errmsg.errorId == "unmatchedSuppression");
-    bool return_value = false;
     for (Suppression &s : mSuppressions) {
```

Comment:
This changes the behavior as it might have matched multiple suppressions before.

Reply:
ah yes this is not intentional. I thought it only wanted to see if there was a match but there is non-const side effects..